File: CollectionViewsObjC/StatesCollectionView_ObjC/StatesCollectionView_ObjC/StateCollectionViewCell.m
//
//  StateCollectionViewCell.m
//  StatesCollectionView_ObjC
//
//  Created by Aaron Martinez on 3/15/17.
//  Copyright © 2017 Aaron Martinez. All rights reserved.
//

#import "StateCollectionViewCell.h"
#import "State.h"

@interface StateCollectionViewCell ()

@property (weak, nonatomic) IBOutlet UILabel *nameLabel;
@property (weak, nonatomic) IBOutlet UIImageView *imageView;

@end

@implementation StateCollectionViewCell

- (void)updateViews {
    self.nameLabel.text = self.state.name;
    self.imageView.image = [UIImage imageNamed: self.state.abbreviation];
}

- (void)setState:(State *)state {
    _state = state;
    [self updateViews];
}

@end

File: CollectionViews/CollectionViews/StateCollectionViewCell.swift
//
//  StateCollectionViewCell.swift
//  CollectionViews
//
//  Created by Spencer Curtis on 3/9/17.
//  Copyright © 2017 Spencer Curtis. All rights reserved.
//

import UIKit

class StateCollectionViewCell: UICollectionViewCell {
    
    @IBOutlet weak var flagImageView: UIImageView!
    @IBOutlet weak var stateNameLabel: UILabel!
    
    var state: State? {
        didSet {
            updateViews()
        }
    }
    
    func updateViews() {
        guard let state = state else { return }
        flagImageView.image = UIImage(named: state.abbreviation)
        stateNameLabel.text = state.name
    }
    
}












